### PR TITLE
[openblas] Update to the 0.3.10

### PR DESCRIPTION
**Pull request by LilyWangL**
Update openblas to version 0.3.10 and fix old patches.

Related #13536

### CODE REVIEW

File: ports/openblas/CONTROL [hunk 1]
```diff
@@ -1,6 +1,5 @@
 Source: openblas
-Version: 0.3.9
-Port-Version: 3
+Version: 0.3.10
 Homepage: https://github.com/xianyi/OpenBLAS
 Build-Depends: pthread (linux&osx)
```
Comment (cenit) [inline ports/openblas/CONTROL:4]: linux&osx ?!?

Comment (ras0219-msft) [reply ports/openblas/CONTROL:4]:
```suggestion
```
`linux&osx` never evaluates to true, so this dependency is doing nothing.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
LGTM after removing the Build-Depends:

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**